# EDA 

* Data from [WiDS Datathon 2021](https://www.kaggle.com/c/widsdatathon2021)

* Objective: To determine whether a patient admitted to an Intensive Care Unit (ICU) has been diagonsed with a particular type of diabetes, Diabetes Mellitus, using data from the first 24 hours of intensive care.

In [1]:
# Import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
sns.set()

from ipywidgets import interact   # interactive plot

In [2]:
# Import training data
train = pd.read_csv("../data/TrainingWiDS2021.csv", index_col=0)

print(train.info())
train.head()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 130157 entries, 1 to 130157
Columns: 180 entries, encounter_id to diabetes_mellitus
dtypes: float64(157), int64(17), object(6)
memory usage: 179.7+ MB
None


encounter_id  hospital_id   age        bmi  elective_surgery  ethnicity  \
1        214826          118  68.0  22.732803                 0  Caucasian   
2        246060           81  77.0  27.421875                 0  Caucasian   
3        276985          118  25.0  31.952749                 0  Caucasian   
4        262220          118  81.0  22.635548                 1  Caucasian   
5        201746           33  19.0        NaN                 0  Caucasian   

  gender  height hospital_admit_source           icu_admit_source  ...  \
1      M   180.3                 Floor                      Floor  ...   
2      F   160.0                 Floor                      Floor  ...   
3      F   172.7  Emergency Department       Accident & Emergency  ...   
4      F   165.1        Operating Room  Operating Room / Recovery  ...   
5      M   188.0                   NaN       Accident & Emergency  ...   

   h1_pao2fio2ratio_max h1_pao2fio2ratio_min aids  cirrhosis  hepatic_failure  \
1                   NaN                  NaN    0          0                0   
2                  51.0                 51.0    0          0                0   
3                   NaN                  NaN    0          0                0   
4                 337.0                337.0    0          0                0   
5                   NaN                  NaN    0          0                0   

   immunosuppression  leukemia  lymphoma  solid_tumor_with_metastasis  \
1                  0         0         0                            0   
2                  0         0         0                            0   
3                  0         0         0                            0   
4                  0         0         0                            0   
5                  0         0         0                            0   

   diabetes_mellitus  
1                  1  
2                  1  
3                  0  
4                  0  
5                  0  

[5 rows x 180 columns]

* Training data: 180 columns

## 1. Training data quality check

In [3]:
# Missing rows (just check, don't do anything yet)
prop_isnull = train.isnull().mean()
prop_isnull = prop_isnull[prop_isnull > 0]
prop_isnull

age                     0.038323
bmi                     0.034497
ethnicity               0.012193
gender                  0.000507
height                  0.015958
                          ...   
h1_arterial_ph_min      0.828607
h1_arterial_po2_max     0.825503
h1_arterial_po2_min     0.825503
h1_pao2fio2ratio_max    0.871232
h1_pao2fio2ratio_min    0.871232
Length: 160, dtype: float64

* 160 out of 180 columns have missing information, number of missing records ranging between 3.8 - 87.1%

In [22]:
# Let's visualize the proportion of missing rows per column

@interact(x=prop_isnull.sort_values().index, y=prop_isnull.sort_values().values)
def get_figure(x, y):
    plt.figure(figsize=(12,6))
    plt.bar(x, y)
    plt.ylabel('Missing values')
    plt.xlabel('Column')
    return plt.show()

interactive(children=(Dropdown(description='x', options=('gender', 'icu_admit_source', 'd1_heartrate_max', 'd1…

In [7]:
# Duplicated rows
train[train.duplicated() == True]

Empty DataFrame
Columns: [encounter_id, hospital_id, age, bmi, elective_surgery, ethnicity, gender, height, hospital_admit_source, icu_admit_source, icu_id, icu_stay_type, icu_type, pre_icu_los_days, readmission_status, weight, albumin_apache, apache_2_diagnosis, apache_3j_diagnosis, apache_post_operative, arf_apache, bilirubin_apache, bun_apache, creatinine_apache, fio2_apache, gcs_eyes_apache, gcs_motor_apache, gcs_unable_apache, gcs_verbal_apache, glucose_apache, heart_rate_apache, hematocrit_apache, intubated_apache, map_apache, paco2_apache, paco2_for_ph_apache, pao2_apache, ph_apache, resprate_apache, sodium_apache, temp_apache, urineoutput_apache, ventilated_apache, wbc_apache, d1_diasbp_invasive_max, d1_diasbp_invasive_min, d1_diasbp_max, d1_diasbp_min, d1_diasbp_noninvasive_max, d1_diasbp_noninvasive_min, d1_heartrate_max, d1_heartrate_min, d1_mbp_invasive_max, d1_mbp_invasive_min, d1_mbp_max, d1_mbp_min, d1_mbp_noninvasive_max, d1_mbp_noninvasive_min, d1_resprate_max, d1_resprate_min, d1_spo2_max, d1_spo2_min, d1_sysbp_invasive_max, d1_sysbp_invasive_min, d1_sysbp_max, d1_sysbp_min, d1_sysbp_noninvasive_max, d1_sysbp_noninvasive_min, d1_temp_max, d1_temp_min, h1_diasbp_invasive_max, h1_diasbp_invasive_min, h1_diasbp_max, h1_diasbp_min, h1_diasbp_noninvasive_max, h1_diasbp_noninvasive_min, h1_heartrate_max, h1_heartrate_min, h1_mbp_invasive_max, h1_mbp_invasive_min, h1_mbp_max, h1_mbp_min, h1_mbp_noninvasive_max, h1_mbp_noninvasive_min, h1_resprate_max, h1_resprate_min, h1_spo2_max, h1_spo2_min, h1_sysbp_invasive_max, h1_sysbp_invasive_min, h1_sysbp_max, h1_sysbp_min, h1_sysbp_noninvasive_max, h1_sysbp_noninvasive_min, h1_temp_max, h1_temp_min, d1_albumin_max, d1_albumin_min, d1_bilirubin_max, d1_bilirubin_min, ...]
Index: []

[0 rows x 180 columns]

* No duplicated rows

## 2. Data distribution

In [8]:
# Data types
train.dtypes.unique()

array([dtype('int64'), dtype('float64'), dtype('O')], dtype=object)

In [9]:
# Data distribution check - numeric / string / categorical ...


In [10]:
# Data cleaning

In [11]:
# Feature engineering

In [12]:
# Modeling - Baseline

In [13]:
# Modeling - Beat the baseline